app/models/subscription.rb

```diff
@@ -63,6 +63,15 @@ def pending_start_date
     (created_at.end_of_month + 1.day).to_date
   end
 
+  def next_pending_start_date
+    following_subscription = next_subscription
+
+    return unless following_subscription
+    return unless following_subscription.pending?
+
+    (following_subscription.created_at.end_of_month + 1.day).to_date
```

Comment:
Just wondering if the `end_of_month` also apply when plan recurrence is yearly?

Reply:
Thanks for the catch! Fixed!